draft-ietf-quic-ack-frequency.md

```diff
@@ -494,9 +494,16 @@ estimate of the typical RTT, but not the minimum RTT (min_rtt)
 unnecessarily high number of acknowledgments when min_rtt is much smaller than
 smoothed_rtt.
 
-Note that the congestion window and the RTT change over the lifetime of a
+Note that the average congestion window and the smoothed RTT change over the lifetime of a
 connection and therefore might require sending frequent ACK_FREQUENCY frames to
-ensure optimal performance.
+ensure optimal performance. As both the congestion window and RTT estimates
+may change continuously, of course not every change should trigger an update.
+Usually, it is not necessary to send an ACK_FREQUENCY frame more than once per
+RTT and likely it needs to be sent even less frequently.
+Ideally, an ACK_FREQUENCY frame is sent only if a relevant change in the
+average congestion window or RTT estimates is detected that impacts the local
+setting of the reordering threshold or locally-selected calculation of the
+Ack-Eliciting Threshold or Requested Max Ack Delay.
```

Comment:
There's that "average" again.

The problem I have with this text is that it says that you compute the value and then only ask the peer to change if the computed value changes.  However, this is still going to thrash quite a bit as estimates fluctuate.  Especially during slow start, where big changes are commonplace (but more ACK frames is fine).

I would instead recommend some amount of hysteresis in values.  That should have the desired effect.

Reply:
The congestion window for most congestion control algorithm makes a sawtooth over a period of cwnd/2 RTTs. The congestion increases by one for cwnd/2 RTT and then is halved when a congestion indication occurs. That means the congestion window changes constantly but the average (over cwnd/2 RTT) stays constant. Only if there is a real change in bandwidth and therefore the average value changes, it makes sense to also adjust the ack frequency.

I tried to change the working to "longer-term development". @martinthomson does that help?

Reply:
I suspect this may just trying to explain too much, not normally sending one more than a RTT seems reasonable; adjusting if there is a significant change in cwnd or smoothed RTT also seems useful to note. I didn't like "average cwnd" or longer term development much though.

Reply:
@gorryfair what do you mean by "not normally sending one more than a RTT "?

Do you have any wording suggestion?